crates/red_knot_python_semantic/src/types.rs

```diff
@@ -498,21 +502,7 @@ pub(crate) use todo_type;
 #[derive(Copy, Clone, Debug, PartialEq, Eq, Hash, salsa::Update)]
 pub enum Type<'db> {
     /// The dynamic type: a statically unknown set of values
-    Any,
-    /// Unknown type (either no annotation, or some kind of type error).
-    /// Equivalent to Any, or possibly to object in strict mode
-    Unknown,
-    /// Temporary type for symbols that can't be inferred yet because of missing implementations.
-    /// Behaves equivalently to `Any`.
-    ///
-    /// This variant should eventually be removed once red-knot is spec-compliant.
-    ///
-    /// General rule: `Todo` should only propagate when the presence of the input `Todo` caused the
-    /// output to be unknown. An output should only be `Todo` if fixing all `Todo` inputs to be not
-    /// `Todo` would change the output type.
-    ///
-    /// This variant should be created with the `todo_type!` macro.
-    Todo(TodoType),
+    Gradual(GradualType),
```

Comment:
So I feel bad suggesting a naming change after you just changed the name for another reviewer... but I would rather call this `Type::Dynamic` and `DynamicType`. "Dynamic" or "dyn" is frequently used for this type in the literature, and usually the term "gradual type" is used as a blanket term encompassing all types, including both fully-static ones and partially-dynamic and fully-dynamic ones. This is also how the Python typing spec uses the term "gradual type."

Reply:
Yeah, this is on me -- I agree `Dynamic` is a better name. Sorry @dcreager!!